### PR TITLE
Upgrate to Flutter 1.12.

### PR DESCRIPTION
**Pull request by ghost**
Upgrade Gradle wrapper to 5.1.1. Migrate example to AndroidX. Clean-up ignored files.

### CODE REVIEW

File: example/ios/Runner/GeneratedPluginRegistrant.m [hunk 1]
```diff
@@ -3,12 +3,10 @@
 //
 
 #import "GeneratedPluginRegistrant.h"
-#import <flutter_auth0/FlutterAuth0Plugin.h>
 
 @implementation GeneratedPluginRegistrant
 
 + (void)registerWithRegistry:(NSObject<FlutterPluginRegistry>*)registry {
-  [FlutterAuth0Plugin registerWithRegistrar:[registry registrarForPlugin:@"FlutterAuth0Plugin"]];
```
Comment (passsy) [inline example/ios/Runner/GeneratedPluginRegistrant.m]: Now it doesn't work on iOS anymore

File: example/.flutter-plugins [hunk 2]
```diff
@@ -1 +1,2 @@
-flutter_auth0=/Users/dennysegura/Dev/Libs/flutter/flutter-auth0/
+# This is a generated file; do not edit or check into version control.
+flutter_auth0=C:\\Dev\\flutter-auth0\\
```
Comment (passsy) [inline example/.flutter-plugins]: That file should be added to .gitignore

File: pubspec.yaml [hunk 3]
```diff
@@ -19,5 +19,7 @@ dev_dependencies:
 
 flutter:
   plugin:
-    androidPackage: io.flutter.plugins.flutterauth0
-    pluginClass: FlutterAuth0Plugin
\ No newline at end of file
+    platforms:
+      android:
```
Comment (passsy) [inline pubspec.yaml]: what about the iOS platform?

File: example/ios/Runner/GeneratedPluginRegistrant.m (hunk 1 above)
Comment (ghost) [reply example/ios/Runner/GeneratedPluginRegistrant.m]: That edit was made automatically when upgrading, I wasn't sure about it, I'll rollback the iOS changes

File: example/.flutter-plugins (hunk 2 above)
Comment (ghost) [reply example/.flutter-plugins]: Will do

File: .gitignore [hunk 4]
```diff
@@ -387,4 +387,9 @@ xcuserdata
...
 .idea/misc.xml
 
-scripts/pub_publish.sh
\ No newline at end of file
+scripts/pub_publish.sh
+.idea/
+.flutter-plugins-dependencies
+.gradle/
+local.properties
+.flutter-plugins
```
Comment (fieldinrain) [inline .gitignore]:
```
.idea/encodings.xml
.idea/misc.xml	.idea/misc.xml
```
can be removed and personally I'd like to add a new line at the EoF .

Comment (ghost) [reply .gitignore]: Updated!

**Pull request merged**